lib/file.js

```diff
@@ -22,8 +22,16 @@ const { HttpStreamError, IllegalArgumentError } = require('./error');
 const { BlockDownload } = require('./block/blockdownload');
 const { BlockUpload } = require('./block/blockupload');
 
+const RETRY_OPTIONS_FILTER = [
+    'retryMaxDuration', 'retryBackoff', 'retryInitialDelay',
+    'retryEnabled', 'retryAllErrors', 'retryMaxCount'
+];
+const CONCURRENCY_OPTIONS_FILTER = [
+    'maxConcurrent', 'concurrent'
+];
+
 /**
- * Check if file path is a directory, or does not exist
+ * Check if file path is a directory, or does not exist (asynchronously)
```

Comment:
Adding the precision about async because some code analysis tool will not detect the returned promises and suggest (wrong) refactorings.

Reply:
I know I kept getting that in my ide :)